DSI/globus_gridftp_server_iRODS.cpp

```diff
@@ -2057,6 +2064,7 @@ globus_l_gfs_iRODS_recv(
                 fd_info = nlohmann::json::parse(j_out);
                 std::string replica_token = fd_info.at("replica_token").get<std::string>();
                 iRODS_handle->replica_token = strdup(replica_token.c_str());
+                free(j_out);
```

Comment:
Is it possible to leak in the error case as well? It may be worth using `irods::at_scope_exit` here.

Also, maybe use `std::free`?

Reply:
Good catch.  Will add the at_scope_exit.

Reply:
I added this in at_scope_exit and changed the calls to std::free().   I did not change older references to free() as that would confuse the review.

I also put nullptr guards around all of these new frees.